pkg/storage/snapshot/restore.go

```diff
@@ -552,10 +554,111 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 	if updated, err := t.reconcileDataVolumes(restoredVM); updated || err != nil {
 		return updated, err
 	}
+	// Reconcile backend storage PVC since it's not part of the VM/VMI spec
+	if ready, err := t.reconcileBackendVolume(snapshotVM); !ready || err != nil {
+		return !ready, err
+	}
 
 	return t.reconcileSpec(restoredVM)
 }
 
+func (t *vmRestoreTarget) reconcileBackendVolume(snapshotVM *snapshotv1.VirtualMachine) (bool, error) {
+	if !backendstorage.IsBackendStorageNeededForVMI(&snapshotVM.Spec.Template.Spec) {
+		return true, nil
+	}
+
+	// Retrieve only the backend volume
+	volumes, err := storageutils.GetVolumes(snapshotVM, t.controller.Client, storageutils.WithBackendVolume)
+	if err != nil {
+		if storageutils.IsErrNoBackendPVC(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	pvc, err := t.controller.getPVC(snapshotVM.Namespace, volumes[0].VolumeSource.PersistentVolumeClaim.ClaimName)
```

Comment:
Would be wise to check the length of `volumes` before accessing

Reply:
ACK

Reply:
Truthfully, I think the cleanest thig to do is just iterate over `volumes`. Why should snapshot layer know there is only one volume?